torch/utils/data/dataloader.py

```diff
@@ -31,6 +31,52 @@ def __init__(self, exc_info):
         self.exc_msg = "".join(traceback.format_exception(*exc_info))
 
 
+# Syscalls are automatically retried upon encountering EINTR since Python 3.5
+# https://www.python.org/dev/peps/pep-0475/
+# EINTR is not available on Windows.
+if sys.platform == 'win32' or sys.version_info >= (3, 5):
```

Comment:
Wouldn't it be cleaner to make our own subclasses of `Queue` such that the code remains unchanged?

Reply:
We can't make a subclass since it deals with three different queue objects: `multiprocessing.Queue`, `multiprocessing.SimpleQueue` and `queue.Queue`. But we can make a wrapper class.

Reply:
Yeah, a wrapper class sounds better